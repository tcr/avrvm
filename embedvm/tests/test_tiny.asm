Label(id='label1')
PushLocal(sfa=-1)
PushConstantV(value=1)
CallUserFunction(funcid=1)
DropValue()
Return0()
PushConstantV(value=35)
CallV(address=LabelRef('label1'))
PopMany(n=0)
DropValue()
PushConstantV(value=0)
CallUserFunction(funcid=0)
DropValue()
Return0()