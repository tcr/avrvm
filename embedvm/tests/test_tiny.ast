  LEFT 0x7fca32c03c70 @ 0000 0000:
    LEFT 0x7fca32c03bb0 @ 0000 0000:
     RIGHT 0x7fca32c03b70 @ 0000 0000:
         LEFT 0x7fca32c03a60 @ 0000 0000: op=3f
        LEFT 0x7fca32c03ab0 @ 0000 0001: op=91
       LEFT 0x7fca32c03af0 @ 0000 0002: op=b1
      RIGHT 0x7fca32c03b30 @ 0000 0003: op=9d
   RIGHT 0x7fca32c03bf0 @ 0000 0004: sym=pin_mode op=9c
 LEFT 0x7fca32c03ff0 @ 0000 0005:
   LEFT 0x7fca32c03f30 @ 0005 0005:
     LEFT 0x7fca32c03de0 @ 0005 0005:
        LEFT 0x7fca32c03cd0 @ 0005 0005: op=98 arg=23
       LEFT 0x7fca32c03d60 @ 0005 0007: op=a2 arg=f9 rel argaddr=0x7fca32c03bf0
        RIGHT 0x7fca32c03d20 @ 0009 0009: op=f8
      RIGHT 0x7fca32c03da0 @ 0005 000a: op=9d
    RIGHT 0x7fca32c03ef0 @ 0005 000b:
       LEFT 0x7fca32c03e30 @ 000b 000b: op=90
      LEFT 0x7fca32c03e70 @ 000b 000c: op=b0
     RIGHT 0x7fca32c03eb0 @ 000b 000d: op=9d
  RIGHT 0x7fca32c03f70 @ 0005 000e: sym=main op=9c
ROOT 0x7fca32c04080 @ 0000 000f:
 RIGHT 0x7fca32c04030 @ 000f 000f: sym=_end
