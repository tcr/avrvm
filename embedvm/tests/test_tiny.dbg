
Total code and data length: 15

pin_mode @ 0000: 3F 91 B1 9D 9C

main @ 0005: 98.23 A2.F9r F8 9D 90 B0 9D 9C

_end @ 000F:

