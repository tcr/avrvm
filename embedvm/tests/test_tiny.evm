function pin_mode (pin) {
  $uf1(pin); // 0 = output, 1 = input
}

function main() {
  pin_mode(35);
  $uf0();
}