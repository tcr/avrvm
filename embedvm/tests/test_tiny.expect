Called user function 1 with 1 args: 5
Main function returned => Terminating.
