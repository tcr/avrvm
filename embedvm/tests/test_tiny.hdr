#define EMBEDVM_SYM__end 0x000f
#define EMBEDVM_SYM_main 0x0005
#define EMBEDVM_SYM_pin_mode 0x0000
#define EMBEDVM_SECT_SRAM_BEGIN 0x0000
#define EMBEDVM_SECT_SRAM_END 0xffff
#define EMBEDVM_SECT_SRAM_DATA 63,145,177,157,156,152,35,162,249,248,157,144,176,157,156
