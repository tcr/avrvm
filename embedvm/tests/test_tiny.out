Called user function 1 with 1 args: 35
Called user function 0 => stop.
